src/main/java/com/cleanroommc/groovyscript/compat/inworldcrafting/Explosion.java

```diff
@@ -0,0 +1,163 @@
+package com.cleanroommc.groovyscript.compat.inworldcrafting;
+
+import com.cleanroommc.groovyscript.GroovyScript;
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.compat.inworldcrafting.jei.ExplosionRecipeCategory;
+import com.cleanroommc.groovyscript.compat.vanilla.VanillaModule;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.cleanroommc.groovyscript.sandbox.ClosureHelper;
+import groovy.lang.Closure;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.fml.common.Optional;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Explosion extends VirtualizedRegistry<Explosion.Recipe> {
+
+    private final List<Recipe> recipes = new ArrayList<>();
+
+    @Optional.Method(modid = "jei")
+    @GroovyBlacklist
+    public List<ExplosionRecipeCategory.RecipeWrapper> getRecipeWrappers() {
+        return this.recipes.stream().map(ExplosionRecipeCategory.RecipeWrapper::new).collect(Collectors.toList());
+    }
+
+    @Override
+    public void onReload() {
+        this.recipes.addAll(getBackupRecipes());
+        getScriptedRecipes().forEach(this.recipes::remove);
+    }
+
+    public void add(Recipe recipe) {
+        this.recipes.add(recipe);
+        addScripted(recipe);
+    }
+
+    public boolean remove(Recipe recipe) {
+        if (this.recipes.remove(recipe)) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public static class Recipe {
+
+        private final IIngredient input;
+        private final ItemStack output;
+        private final float chance;
+        // modifier to approximate normal distribution
+        private final float statisticalModifier;
+        private final Closure<Boolean> beforeRecipe;
+
+        public Recipe(IIngredient input, ItemStack output, float chance, Closure<Boolean> beforeRecipe) {
+            this.input = input;
+            this.output = output;
+            this.chance = chance;
+            // const value based on e^(-x^2)
+            this.statisticalModifier = (float) (Math.pow(1_000_000, (chance - 0.5f) * (0.5f - chance)) * 0.3f);
```

Comment:
Should this be a modifiable function that takes `chance`?

Reply:
I don't see the use case. This just a const value to simulate randomness without doing `rnd.nextBoolean()` `stack.getCount()` amount of times.